modules/zip/src/hash-file-utility.ts

```diff
@@ -77,3 +82,26 @@ export const findBin = (
   }
   return undefined;
 };
+
+/**
+ * generates hash info from central directory
+ * @param fileProvider - provider of the archive
+ * @returns ready to use hash info
+ */
+export const generateHashInfo = async (fileProvider: FileProvider): Promise<HashElement[]> => {
+  const {cdStartOffset} = await parseEoCDRecord(fileProvider);
+  let cdHeader = await parseZipCDFileHeader(cdStartOffset, fileProvider);
+  const hashInfo: HashElement[] = [];
+  while (cdHeader) {
+    hashInfo.push({
+      hash: Buffer.from(md5(cdHeader.fileName.split('\\').join('/').toLocaleLowerCase()), 'hex'),
+      offset: cdHeader.localHeaderOffset
+    });
+    cdHeader = await parseZipCDFileHeader(
```

Comment:
Nit: I would probably have split this in two parts. Make one function that returns the list of headers and another that takes that list and generates hashes. Now there is a lot going on in here.